src/EditorFeatures/CSharp/Completion/CompletionProviders/XmlDocCommentCompletion/XmlDocCommentCompletionProvider.cs

```diff
@@ -220,15 +220,39 @@ private string AttributeSelector(XmlElementSyntax element, string attribute)
             return null;
         }
 
-        private IEnumerable<CompletionItem> GetTagsForProperty(IPropertySymbol symbol, TextSpan filterSpan, DocumentationCommentTriviaSyntax trivia)
+        private IEnumerable<CompletionItem> GetTagsForProperty(IPropertySymbol symbol, TextSpan filterSpan, DocumentationCommentTriviaSyntax trivia, SyntaxToken token)
         {
             var items = new List<CompletionItem>();
 
-            var typeParameters = symbol.GetTypeArguments().Select(p => p.Name).ToSet();
+            if (symbol.IsIndexer)
+            {
+                var parameters = symbol.GetParameters().Select(p => p.Name).ToSet();
 
-            RemoveExistingTags(trivia, typeParameters, x => AttributeSelector(x, "typeparam"));
+                // User is trying to write a name, try to suggest only names.
+                if (token.Parent.IsKind(SyntaxKind.XmlNameAttribute) ||
+                    (token.Parent.IsKind(SyntaxKind.IdentifierName) && token.Parent.IsParentKind(SyntaxKind.XmlNameAttribute)))
```

Comment:
Thats a lot of parameters. To sum that up: 
- Add a 
```csharp
 private IEnumerable<CompletionItem> GetParameterNameTags(TextSpan filterSpan, SyntaxToken token, List<CompletionItem> items, ISet<string> parameters, ISet<string> typeParameters)
```
- Call that method both in `GetTagsForProperty` and `GetTagsForMethod`.
- Use ContainingType.Typeparameters for the type parameters in the GetTagsForProperty and ContainingType.Typeparameters + the method type parameters for the type parameters in GetTagsForMethod

Im asking because typeparamref doesnt have completion for containing type type parameters yet.